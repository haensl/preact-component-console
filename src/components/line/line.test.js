import { h } from 'preact';
import render from 'preact-render-to-string';
import Line, { DEFAULTS } from './';

describe('Line', () => {
  describe('empty', () => {
    let tree;

    beforeEach(() => {
      tree = render(<Line />);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('adds the default element class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}"`));
    });
  });

  describe('ps1', () => {
    let tree;

    beforeEach(() => {
      tree = render(<Line ps1={ {
        content: 'test'
      }} />);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('sets the content of ps1 correctly', () => {
      expect(tree).toMatch(/>test</);
    });

    it('adds the default element class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}"`));
    });
  });

  describe('content', () => {
    let tree;

    beforeEach(() => {
      tree = render(<Line content="foo bar" />);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('sets the content correctly', () => {
      expect(tree).toMatch(/>foo bar</);
    });

    it('adds the default element class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}"`));
    });
  });

  describe('class', () => {
    let tree;

    describe('element', () => {
      beforeEach(() => {
        tree = render(<Line classes={{
          element: 'testclass'
        }} />);
      });

      it('renders as expected', () => {
        expect(tree).toMatchSnapshot();
      });

      it('sets the class correctly', () => {
        expect(tree).toMatch(/class="testclass"/);
      });
    });

    describe('content', () => {
      beforeEach(() => {
        tree = render(<Line classes={{
          content: 'testcontent'
        }} />);
      });

      it('renders as expected', () => {
        expect(tree).toMatchSnapshot();
      });

      it('sets the class correctly', () => {
        expect(tree).toMatch(/class="testcontent"/);
      });
    });
  });

  describe('write', () => {
    let tree;

    beforeEach(() => {
      tree = render(<Line write={ true } cursor={{
        char: 't'
      }}/>);
    });

    it('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    it('sets the char under the cursor correctly', () => {
      expect(tree).toMatch(/>t</);
    });

    it('adds the default element class', () => {
      expect(tree).toMatch(new RegExp(`class="${DEFAULTS.classes.element}"`));
    });
  });
});
