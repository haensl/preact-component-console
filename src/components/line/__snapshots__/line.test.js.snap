// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line class content renders as expected 1`] = `"<pre class=\\"console-line\\"><span class=\\"testcontent\\"></span></pre>"`;

exports[`Line class element renders as expected 1`] = `"<pre class=\\"testclass\\"><span class=\\"console-line-content\\"></span></pre>"`;

exports[`Line content renders as expected 1`] = `"<pre class=\\"console-line\\"><span class=\\"console-line-content\\">foo bar</span></pre>"`;

exports[`Line empty renders as expected 1`] = `"<pre class=\\"console-line\\"><span class=\\"console-line-content\\"></span></pre>"`;

exports[`Line ps1 renders as expected 1`] = `"<pre class=\\"console-line\\"><span class=\\"console-ps1\\">test</span><span class=\\"console-line-content\\"></span></pre>"`;

exports[`Line write renders as expected 1`] = `"<pre class=\\"console-line\\"><span class=\\"console-line-content\\"></span><span class=\\"console-cursor console-cursor--write\\">t</span></pre>"`;
